Index: public/sdk/v4/p.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- public/sdk/v4/p.js	(revision e3b8838172645a4e0cbcfa11eb67055d5b4cfa50)
+++ public/sdk/v4/p.js	(revision )
@@ -912,6 +912,26 @@
             }
 
             /**
+             * Get only domain name used for cookie.  For example, host is "secured.host.com" then
+             * this function will return ".host.com".
+             *
+             * @param host {string] complete host name
+             * @returns {string}
+             */
+            function getOnlyDomain(host) {
+                if (host) {
+                    var s = host.split('.');
+                    if (s.length > 2) {
+                        s.shift();
+                        return "." + s.join('.');
+                    } else {
+                        return "." + host;
+                    }
+                }
+                return host? host: '';
+            }
+
+            /**
              * Set cookie value
              * @params {string} cookie name
              * @params {string} value
@@ -936,8 +956,8 @@
 
                 document_alias.cookie = cookie_name + '=' + encode_wrapper(value) +
                         (ms_to_expire ? ';expires=' + expiry_date.toGMTString() : '') +
-                        ';path=' + (path || '/') +
-                        (domain ? ';domain=' + domain : '') +
+                        ';path=' + (path || '/') + ';domain=' +
+                        (domain ? domain : getOnlyDomain(window.location.hostname)) +
                         (secure ? ';secure' : '');
             }
 
